test/replication-luatest/gh_7512_raft_leader_hang_test.lua

```diff
@@ -0,0 +1,67 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local fio = require('fio')
+
+local g = t.group('gh-7512')
+
+
+-- gh-7512: Raft followers should notice leader hang and arrange new elections.
+g.before_all(function(cg)
+    cg.cluster = cluster:new{}
+    local box_cfg = {
+        election_mode = 'candidate',
+        replication_timeout = 0.1,
+        replication_synchro_quorum = 1,
+        replication = {
+            server.build_instance_uri('node1'),
+            server.build_instance_uri('node2'),
+        },
+    }
+    cg.node1 = cg.cluster:build_and_add_server{
+        alias = 'node1',
+        box_cfg = box_cfg,
+    }
+    box_cfg.election_mode='voter'
+    cg.node2  = cg.cluster:build_and_add_server{
+        alias = 'node2',
+        box_cfg = box_cfg,
+    }
+    cg.cluster:start()
+
+    cg.unique_filename = server.build_instance_uri('unique_filename')
+    fio.unlink(cg.unique_filename)
+end)
+
+g.after_all(function(cg)
+    cg.cluster:drop()
+    fio.unlink(cg.unique_filename)
+end)
+
+local function block(name)
+    local fiber = require('fiber')
+    fiber.new(function(name)
+        -- Can't use fio, because it yields.
+        while os.execute('ls ' .. name) ~= 0 do end
```

Comment:
I've run the test myself directly with `luatest -c` and noticed unneeded output "node1 | ls: cannot access '/tmp/t/unique_filename.iproto': No such file or directory" many times.
changing this line to `while os.execute('sleep 0.01 && ls ' .. name .. ' > /dev/null 2>&1') ~= 0 do end` works even better, suppressing useless output.

Reply:
Thanks for the suggestion! Fixed.
```diff
diff --git a/test/replication-luatest/gh_7512_raft_leader_hang_test.lua b/test/replication-luatest/gh_7512_raft_leader_hang_test.lua
index 50f41452a..5ad60e98a 100644
--- a/test/replication-luatest/gh_7512_raft_leader_hang_test.lua
+++ b/test/replication-luatest/gh_7512_raft_leader_hang_test.lua
@@ -39,12 +39,13 @@ g.after_all(function(cg)
     fio.unlink(cg.unique_filename)
 end)
 
-local function block(name)
+local function block(filename)
     local fiber = require('fiber')
-    fiber.new(function(name)
+    fiber.new(function(filename)
         -- Can't use fio, because it yields.
-        while os.execute('ls ' .. name) ~= 0 do end
-    end, name)
+        local cmd = 'sleep 0.1 && ls ' .. filename .. ' > /dev/null 2>&1'
+        while os.execute(cmd) ~= 0 do end
+    end, filename)
 end
 
 g.test_leader_hang_notice = function(cg)
```